modules/tensor_mechanics/src/materials/ADComputeFiniteStrainElasticStress.C

```diff
@@ -16,15 +16,26 @@ ADComputeFiniteStrainElasticStress::validParams()
 {
   InputParameters params = ADComputeStressBase::validParams();
   params.addClassDescription("Compute stress using elasticity for finite strains");
+  params.addParam<bool>(
+      "use_old_elasticity_tensor",
+      false,
+      "Flag to optionally use the elasticity tensor computed at the previous timestep.");
   return params;
 }
 
 ADComputeFiniteStrainElasticStress::ADComputeFiniteStrainElasticStress(
     const InputParameters & parameters)
   : ADComputeStressBase(parameters),
     GuaranteeConsumer(this),
+    _use_old_elasticity_tensor(getParam<bool>("use_old_elasticity_tensor")),
     _elasticity_tensor_name(_base_name + "elasticity_tensor"),
-    _elasticity_tensor(getADMaterialProperty<RankFourTensor>(_elasticity_tensor_name)),
+    _elasticity_tensor(
+        _use_old_elasticity_tensor
+            ? getGenericZeroMaterialProperty<RankFourTensor, true>(_elasticity_tensor_name)
```

Comment:
Once #16000 is in this should be
```suggestion
            ? getGenericZeroMaterialProperty<RankFourTensor, true>()
```

Reply:
Chip chop chip

Reply:
Otherwise it will pull in the elasticity tensor! You can probably just get that tensor indiscriminately as there is no penalty. Where it matters is possibly the old state, because the elasticity tensor might not be stateful.

Reply:
![image](https://user-images.githubusercontent.com/7014706/97607345-77e31a80-19d6-11eb-9e10-109e8ed861a7.png)

Reply:
I'm just being difficult. All of these have to be changed after #16000, thanks for the reminder.